contracts/RootChain.sol

```diff
@@ -292,9 +292,14 @@ contract RootChain {
         // deposit with blknum ending with 000.
         require(nextDepositBlock < CHILD_BLOCK_INTERVAL);
 
-        // Check that all but first outputs are 0.
-        for (uint i = 1; i < MAX_INPUTS; i++) {
-            require(decodedTx.outputs[i].amount == 0);
+        for (uint i = 0; i < MAX_INPUTS; i++) {
+            // all inputs should be empty
+            require(decodedTx.inputs[i].blknum == 0);
```

Comment:
checking only `blknum` is safe as we start counting blocks from 1, so even if other fields would not equal to 0, the deposit transaction would not be spending any valid `utxo`

Reply:
please note that the deposit transaction itself is not signed, so it cannot be used to challenge any other transaction (shared inputs)

Reply:
ouch, didn't look clear enough from the issue lol and misunderstood the intention. I will change my PR description and commit message before merge

Reply:
Sorry, the issue was rather a quick note. I meant to document it better, but you took it earlier ;)